### PR TITLE
RTMD-34 Updated the session middleware in app.js to prevent a memory …

### PR DESCRIPTION
**Pull request by bendiggle**
…leak

### CODE REVIEW

**Comment by JoeChapman** [comment]
Thanks @bendiggle but have a look here: https://github.com/UKHomeOffice/hof-bootstrap/blob/master/lib/sessions.js

Perhaps some lessons we can take from it.

Also: @easternbloc, @joefitter

**Comment by easternbloc** [comment]
Please cite other PR's if you're taking guidance from them. It's good for whoever is looking at the codes to have as much insight as possible.

Can you add the fixing PR to the commit message: https://github.com/UKHomeOffice/brp_app/pull/329 ❤️

**Comment by easternbloc** [comment]
And yes @JoeChapman we can i'm sure

**Comment by bendiggle** [comment]
@easternbloc How does the commit message look now?
@JoeChapman This has been updated now

**Comment by JoeChapman** [comment]
LGTM

**Comment by easternbloc** [comment]
👍

File: app.js [hunk 1]
```diff
@@ -79,19 +79,16 @@ function secureCookies(req, res, next) {
...
-    },
-    name: config.session.name,
-    secret: config.session.secret,
-    resave: true,
-    saveUninitialized: true
-  })(req, res, next);
-}
-app.use(initSession);
+const sessionOpts = Object.assign({
+  redisStore,
```
Comment (joefitter) [inline app.js]: @bendiggle this should be `store: redisStore`

Comment (easternbloc) [inline app.js]: ah crap

Comment (easternbloc) [inline app.js]: yes, yes it should

**Comment by easternbloc** [comment]
A bigger issue here. Why did the tests pass?

**Comment by joefitter** [comment]
because the tests don't restart the server

**Comment by joefitter** [comment]
data is persisted for the lifecycle of the application just as a general model.

**Comment by easternbloc** [comment]
ah not good :(

**Pull request merged**